backend/src/main/java/ro/utcn/sd/boti/stackoverflow/controller/Controller.java

```diff
@@ -0,0 +1,70 @@
+package ro.utcn.sd.boti.stackoverflow.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.event.EventListener;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.sd.boti.stackoverflow.dto.AnswerDTO;
+import ro.utcn.sd.boti.stackoverflow.dto.QuestionDTO;
+import ro.utcn.sd.boti.stackoverflow.dto.SUserDTO;
+import ro.utcn.sd.boti.stackoverflow.event.BaseEvent;
+import ro.utcn.sd.boti.stackoverflow.service.AnswerService;
+import ro.utcn.sd.boti.stackoverflow.service.QuestionService;
+import ro.utcn.sd.boti.stackoverflow.service.SUserService;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+public class Controller {
+    private final QuestionService questionService;
+    private final SUserService sUserService;
+    private final AnswerService answerService;
+    private final SimpMessagingTemplate messagingTemplate;
+
+    @GetMapping("/questions")
+    public List<QuestionDTO> readAllQuestions() {
+        return questionService.listQuestions();
+    }
+
+    @PostMapping("/questions")
+    public QuestionDTO createQuestion(@RequestBody QuestionDTO dto) {
+        return questionService.addQuestion(dto);
+    }
+
+    @GetMapping("/filter-by-text/{text}")
+    public List<QuestionDTO> filterByText(@PathVariable String text) {
+        return questionService.searchInTitle(text);
+    }
+
+    @GetMapping("/filter-by-tag/{tag}")
```

Comment:
Would have been more appropriate to use the `questions` path with `@RequestParam`s. https://www.baeldung.com/spring-request-param